src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -453,6 +453,12 @@ private function addRouterSection(ArrayNodeDefinition $rootNode)
                             ->defaultTrue()
                         ->end()
                         ->booleanNode('utf8')->defaultFalse()->end()
+                        ->scalarNode('query_enc_type')
+                            ->info(
+                                "set to PHP_QUERY_RFC1738 to make the UrlGenerator encode spaces as plus signs (+) when building query strings\n"
```

Comment:
This info is only correct for the default value, it should probably be something more generic.

How many possibilities are there for this option? Because it would be really nice if Symfony would tell you when using an invalid value. Perhaps you can limit it to the possible methods using validation?